Explorer/Assets/DCL/SDKComponents/AvatarShape/Systems/AvatarShapeHandlerSystem.cs

```diff
@@ -89,7 +100,108 @@ private void UpdateAvatarShape(ref PBAvatarShape pbAvatarShape, ref SDKAvatarSha
             globalWorld.Set(sdkAvatarShapeComponent.globalWorldEntity, pbAvatarShape);
 
             if (!string.IsNullOrEmpty(pbAvatarShape.ExpressionTriggerId))
-                globalWorld.AddOrSet(sdkAvatarShapeComponent.globalWorldEntity, new CharacterEmoteIntent() { EmoteId = pbAvatarShape.ExpressionTriggerId });
+                AddCharacterEmoteIntent(sdkAvatarShapeComponent.globalWorldEntity, pbAvatarShape.ExpressionTriggerId, BodyShape.FromStringSafe(pbAvatarShape.BodyShape));
+        }
+
+        private void AddCharacterEmoteIntent(Entity globalWorldEntity, string emoteId, BodyShape bodyShape)
+        {
+            bool isSceneEmote = emoteId.EndsWith(".glb", StringComparison.OrdinalIgnoreCase);
+            if (!isSceneEmote) // normal "urn emote" or "base emote"
+            {
+                globalWorld.AddOrSet(globalWorldEntity,
+                    new CharacterEmoteIntent { EmoteId = emoteId });
+
+                return;
+            }
+
+            if (!sceneData.SceneContent.TryGetHash(emoteId, out string hash))
+            {
+                ReportHub.LogError(ReportCategory.AVATAR,$"Scene emote '{emoteId}' not found in scene assets. Aborting scene emote playing on SDK AvatarShape");
+                return;
+            }
+
+            if (globalWorld.TryGet(globalWorldEntity, out SDKAvatarShapeEmotePromiseCancellationToken? promiseComponent))
+                promiseComponent!.Dispose();
+
+            var newPromiseCancellationTokenComponent = new SDKAvatarShapeEmotePromiseCancellationToken();
```

Comment:
What was the reason to break the existing scheme for promises?

E.g. here you don't need a separate cancellation component as when you create an intention the cancellation token source will be already there